internal/network/none_platform_node_ips_allocation.go

```diff
@@ -0,0 +1,312 @@
+package network
+
+import (
+	"encoding/json"
+	stderrors "errors"
+	"net"
+	"sort"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+	"github.com/samber/lo"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+// When none platform is in use, if there is ambiguity in node-ip assignment, then incorrect assignment might lead
+// to installation failure.  This happens when etcd detects that the socket address from an etcd node does not match
+// the expected address in the peer certificate. In this case etcd rejects such connection.
+// Example: assuming two networks - net1 and net2.
```

Comment:
What will happen in the following scenario?
assuming two networks - net1 and net2.
// master node 1 has 1 address that belongs to net1.
// master node 2 has 2 addresses.  one that belongs to net 1, and another that belongs to net 2
// master node 3 has 1 address that belongs to net2.

Reply:
This is a failure to allocate.
According to my tests, the installation will fail too.
If we want, we can use this as a pre-install validation